lib/sqlStoreStategy.js

```diff
@@ -0,0 +1,107 @@
+
+class SQLStoreStategy {
+
+  constructor(pool, table) {
+    this.pool = pool;
+    this.table = table;
+    this.dbValidated = false;
+  }
+
+  async validateDB() {
+    if (this.dbValidated) return true;
+    await this.pool.query(
+      `CREATE TABLE IF NOT EXISTS ${this.table} ` +
+      '(id varchar primary key, value varchar);'
+    );
+    this.dbValidated = true;
+    return true;
+  }
+
+  validateKey(key) {
+    if (key && key.length === 0) {
```

Comment:
This means that "undefined" and "0" are actually valid keys. If we want to adhere to my doc then you need something like:

if (typeof key !== 'string' || key.length === 0) {
  throw new Error(`Datastore: Invalid key: ${key}`);
}

(adding the key to the exception helps later when debugging)

Reply:
good point